__tests__/app.test.js

```diff
@@ -28,37 +29,45 @@ describe("getTopics", () => {
         .get("/api")
         .expect(200)
         .then(({ body }) => {
-          expect(body).toEqual({
-            "GET /api": {
-              description:
-                "serves up a json representation of all the available endpoints of the api",
-            },
-            "GET /api/topics": {
-              description: "serves an array of all topics",
-              queries: [],
-              exampleResponse: {
-                topics: [{ slug: "football", description: "Footie!" }],
-              },
-            },
-            "GET /api/articles": {
-              description: "serves an array of all articles",
-              queries: ["author", "topic", "sort_by", "order"],
-              exampleResponse: {
-                articles: [
-                  {
-                    title: "Seafood substitutions are increasing",
-                    topic: "cooking",
-                    author: "weegembump",
-                    body: "Text from the article..",
-                    created_at: "2018-05-30T15:59:13.341Z",
-                    votes: 0,
-                    comment_count: 6,
-                  },
-                ],
-              },
-            },
-          });
+          expect(body).toEqual(endpoint);
         });
     });
   });
 });
+
+describe('getArticles', () => {
+    test('200: /api/articles/:article_id', () => {
+        return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        body.forEach((column) => {
+          expect(column).toMatchObject({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+          })
+        })
+      });
+    });
+    test('400: should return bad request ', () => {
+        return request(app)
+        .get("/api/articles/NaN")
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe(`ERROR: bad request. use "/api" for acceptable endpoints`)

```

Comment:
With this error message, I'd change the description slightly. In this instance the use would've used /api in the get request so it's slightly misleading. It should perhaps say something like "ensure you use a valid article ID number" 👍